{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0c9a5a92",
   "metadata": {
    "papermill": {
     "duration": 0.017475,
     "end_time": "2022-11-29T18:40:53.916201",
     "exception": false,
     "start_time": "2022-11-29T18:40:53.898726",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f8d553dd",
   "metadata": {
    "papermill": {
     "duration": 0.010328,
     "end_time": "2022-11-29T18:40:53.937391",
     "exception": false,
     "start_time": "2022-11-29T18:40:53.927063",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "574f2e07",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:40:53.992267Z",
     "iopub.status.busy": "2022-11-29T18:40:53.991961Z",
     "iopub.status.idle": "2022-11-29T18:41:01.220089Z",
     "shell.execute_reply": "2022-11-29T18:41:01.219279Z"
    },
    "papermill": {
     "duration": 7.276031,
     "end_time": "2022-11-29T18:41:01.223452",
     "exception": false,
     "start_time": "2022-11-29T18:40:53.947421",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "788bf340",
   "metadata": {
    "papermill": {
     "duration": 0.01079,
     "end_time": "2022-11-29T18:41:01.250438",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.239648",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5aec8d7d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:01.273061Z",
     "iopub.status.busy": "2022-11-29T18:41:01.272822Z",
     "iopub.status.idle": "2022-11-29T18:41:01.278287Z",
     "shell.execute_reply": "2022-11-29T18:41:01.277748Z"
    },
    "papermill": {
     "duration": 0.018639,
     "end_time": "2022-11-29T18:41:01.279809",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.261170",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'ana-flavia-superlactacao.yaml')\n",
    "experiment_name = 'jenneffer-vs-01'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'redemicro')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'not-hist-vs-manifest.csv')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'imgs'))\n",
    "replace_files = False\n",
    "trim = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "95dc44d6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:01.303521Z",
     "iopub.status.busy": "2022-11-29T18:41:01.303284Z",
     "iopub.status.idle": "2022-11-29T18:41:01.307622Z",
     "shell.execute_reply": "2022-11-29T18:41:01.307073Z"
    },
    "papermill": {
     "duration": 0.017354,
     "end_time": "2022-11-29T18:41:01.309154",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.291800",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"andressa\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-andressa-lbtm\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-andressa-lbtm/data/manifest.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-andressa-lbtm/data/metadata.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f0b6686c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.610682Z",
     "start_time": "2022-06-28T18:50:41.595886Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:01.332607Z",
     "iopub.status.busy": "2022-11-29T18:41:01.332327Z",
     "iopub.status.idle": "2022-11-29T18:41:01.338399Z",
     "shell.execute_reply": "2022-11-29T18:41:01.337440Z"
    },
    "papermill": {
     "duration": 0.020832,
     "end_time": "2022-11-29T18:41:01.340746",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.319914",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Setting hardcoded parameters for development and debbuging\n",
    "production = True\n",
    "if not production:\n",
    "    params_path = os.path.join('..', 'params', 'ana-flavia-hipotese-01')\n",
    "    with open(params_path, 'r') as stream:\n",
    "        params = yaml.safe_load(stream)\n",
    "        experiment_name = params['experiment_name']\n",
    "        base_dir = params['base_dir']\n",
    "        manifest_file = params['manifest_file']\n",
    "        replace_files = params['replace_files']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e37b45ca",
   "metadata": {
    "papermill": {
     "duration": 0.01074,
     "end_time": "2022-11-29T18:41:01.364478",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.353738",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1df5ac15",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.618098Z",
     "start_time": "2022-06-28T18:50:41.614152Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:01.386681Z",
     "iopub.status.busy": "2022-11-29T18:41:01.386449Z",
     "iopub.status.idle": "2022-11-29T18:41:01.389545Z",
     "shell.execute_reply": "2022-11-29T18:41:01.389025Z"
    },
    "papermill": {
     "duration": 0.015859,
     "end_time": "2022-11-29T18:41:01.390975",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.375116",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# new_manifest = '/home/lauro/nupeb/redemicro/data/raw/manifest/karina-manifest.csv'\n",
    "# with open(manifest_file, 'r') as oldm, open(new_manifest, 'w') as newm:\n",
    "#     header = 'sample-id,absolute-filepath,direction\\n'\n",
    "#     newm.write(header)\n",
    "#     for line in oldm.readlines()[1:]:\n",
    "#         sid, forward, reverse = line[:-1].split(',')\n",
    "#         fline = ','.join((sid, forward, 'forward')) + '\\n'\n",
    "#         rline = ','.join((sid, reverse, 'reverse')) + '\\n'\n",
    "#         newm.write(fline)\n",
    "#         newm.write(rline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7af7c8df",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:01.413072Z",
     "iopub.status.busy": "2022-11-29T18:41:01.412850Z",
     "iopub.status.idle": "2022-11-29T18:41:01.417882Z",
     "shell.execute_reply": "2022-11-29T18:41:01.417393Z"
    },
    "papermill": {
     "duration": 0.017735,
     "end_time": "2022-11-29T18:41:01.419323",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.401588",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e59feee",
   "metadata": {
    "papermill": {
     "duration": 0.010544,
     "end_time": "2022-11-29T18:41:01.441060",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.430516",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3b99147b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:01.463230Z",
     "iopub.status.busy": "2022-11-29T18:41:01.463006Z",
     "iopub.status.idle": "2022-11-29T18:41:01.470711Z",
     "shell.execute_reply": "2022-11-29T18:41:01.469919Z"
    },
    "papermill": {
     "duration": 0.020402,
     "end_time": "2022-11-29T18:41:01.472137",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.451735",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "95812c7e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:01.496554Z",
     "iopub.status.busy": "2022-11-29T18:41:01.496397Z",
     "iopub.status.idle": "2022-11-29T18:41:10.925945Z",
     "shell.execute_reply": "2022-11-29T18:41:10.925261Z"
    },
    "papermill": {
     "duration": 9.44361,
     "end_time": "2022-11-29T18:41:10.927821",
     "exception": false,
     "start_time": "2022-11-29T18:41:01.484211",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        d_type, \n",
    "        manifest_file, \n",
    "        view_type=v_type)\n",
    "    \n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "97db7e38",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:11.127707Z",
     "iopub.status.busy": "2022-11-29T18:41:11.127400Z",
     "iopub.status.idle": "2022-11-29T18:41:30.680704Z",
     "shell.execute_reply": "2022-11-29T18:41:30.679944Z"
    },
    "papermill": {
     "duration": 19.743342,
     "end_time": "2022-11-29T18:41:30.683719",
     "exception": false,
     "start_time": "2022-11-29T18:41:10.940377",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b145d5b",
   "metadata": {
    "papermill": {
     "duration": 0.014401,
     "end_time": "2022-11-29T18:41:30.716286",
     "exception": false,
     "start_time": "2022-11-29T18:41:30.701885",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5fb41bc1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:30.756872Z",
     "iopub.status.busy": "2022-11-29T18:41:30.756638Z",
     "iopub.status.idle": "2022-11-29T18:41:30.760993Z",
     "shell.execute_reply": "2022-11-29T18:41:30.760334Z"
    },
    "papermill": {
     "duration": 0.025229,
     "end_time": "2022-11-29T18:41:30.762482",
     "exception": false,
     "start_time": "2022-11-29T18:41:30.737253",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: 1e4f5d56-f644-4b61-8958-8112d6dc3b95>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "efcb9baf",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:30.785083Z",
     "iopub.status.busy": "2022-11-29T18:41:30.784921Z",
     "iopub.status.idle": "2022-11-29T18:41:30.840192Z",
     "shell.execute_reply": "2022-11-29T18:41:30.839133Z"
    },
    "papermill": {
     "duration": 0.068252,
     "end_time": "2022-11-29T18:41:30.841958",
     "exception": false,
     "start_time": "2022-11-29T18:41:30.773706",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-8dlenpz6')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 1e4f5d56-f644-4b61-8958-8112d6dc3b95>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b3c914b9",
   "metadata": {
    "papermill": {
     "duration": 0.011293,
     "end_time": "2022-11-29T18:41:30.864983",
     "exception": false,
     "start_time": "2022-11-29T18:41:30.853690",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2254093d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-11-29T18:41:30.888872Z",
     "iopub.status.busy": "2022-11-29T18:41:30.888512Z",
     "iopub.status.idle": "2022-11-29T18:42:21.419410Z",
     "shell.execute_reply": "2022-11-29T18:42:21.418868Z"
    },
    "papermill": {
     "duration": 50.546374,
     "end_time": "2022-11-29T18:42:21.422545",
     "exception": false,
     "start_time": "2022-11-29T18:41:30.876171",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S62_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S62_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S62_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S62_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S62_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S62_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S62_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S62_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.14 s (18 µs/read; 3.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            119,387\n",
      "  Read 1 with adapter:                 118,446 (99.2%)\n",
      "  Read 2 with adapter:                 117,302 (98.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,008 (2.5%)\n",
      "Pairs written (passing filters):       116,379 (97.5%)\n",
      "\n",
      "Total basepairs processed:    60,887,370 bp\n",
      "  Read 1:    36,413,035 bp\n",
      "  Read 2:    24,474,335 bp\n",
      "Total written (filtered):     54,191,998 bp (89.0%)\n",
      "  Read 1:    33,051,306 bp\n",
      "  Read 2:    21,140,692 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 384 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 8.6%\n",
      "  G: 90.4%\n",
      "  T: 1.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t12\t1.8\t1\t0 12\n",
      "9\t10\t0.5\t1\t0 10\n",
      "10\t6\t0.1\t1\t0 6\n",
      "18\t1\t0.0\t2\t1\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t3\t0.0\t3\t3\n",
      "22\t4\t0.0\t3\t4\n",
      "23\t6\t0.0\t3\t5 1\n",
      "24\t9\t0.0\t3\t8 0 1\n",
      "25\t6\t0.0\t3\t5 1\n",
      "26\t1\t0.0\t3\t1\n",
      "28\t2\t0.0\t3\t2\n",
      "31\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "39\t1\t0.0\t3\t1\n",
      "58\t1\t0.0\t3\t1\n",
      "84\t1\t0.0\t3\t1\n",
      "86\t1\t0.0\t3\t1\n",
      "88\t1\t0.0\t3\t1\n",
      "96\t1\t0.0\t3\t1\n",
      "101\t2\t0.0\t3\t2\n",
      "107\t1\t0.0\t3\t1\n",
      "108\t1\t0.0\t3\t1\n",
      "113\t1\t0.0\t3\t1\n",
      "119\t1\t0.0\t3\t1\n",
      "132\t1\t0.0\t3\t1\n",
      "138\t2\t0.0\t3\t2\n",
      "148\t1\t0.0\t3\t1\n",
      "149\t1\t0.0\t3\t1\n",
      "151\t1\t0.0\t3\t1\n",
      "152\t1\t0.0\t3\t1\n",
      "155\t1\t0.0\t3\t1\n",
      "160\t1\t0.0\t3\t0 1\n",
      "163\t1\t0.0\t3\t1\n",
      "165\t1\t0.0\t3\t1\n",
      "171\t1\t0.0\t3\t1\n",
      "195\t1\t0.0\t3\t1\n",
      "206\t1\t0.0\t3\t1\n",
      "216\t1\t0.0\t3\t1\n",
      "217\t2\t0.0\t3\t2\n",
      "220\t1\t0.0\t3\t0 0 1\n",
      "221\t1\t0.0\t3\t1\n",
      "223\t1\t0.0\t3\t1\n",
      "226\t1\t0.0\t3\t1\n",
      "229\t1\t0.0\t3\t1\n",
      "231\t1\t0.0\t3\t1\n",
      "233\t35\t0.0\t3\t33 2\n",
      "234\t48\t0.0\t3\t47 1\n",
      "235\t57\t0.0\t3\t55 2\n",
      "236\t53\t0.0\t3\t52 1\n",
      "237\t39\t0.0\t3\t38 0 1\n",
      "238\t47\t0.0\t3\t47\n",
      "239\t2\t0.0\t3\t2\n",
      "243\t1\t0.0\t3\t1\n",
      "249\t1\t0.0\t3\t1\n",
      "252\t1\t0.0\t3\t1\n",
      "264\t2\t0.0\t3\t1 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 118062 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.5\t1\t0 1\n",
      "10\t3\t0.1\t1\t2 1\n",
      "11\t1\t0.0\t1\t1\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t11\t0.0\t2\t9 2\n",
      "15\t10\t0.0\t2\t8 1 1\n",
      "16\t7\t0.0\t2\t5 1 1\n",
      "17\t66\t0.0\t2\t33 23 10\n",
      "18\t18945\t0.0\t2\t18626 304 15\n",
      "19\t21148\t0.0\t2\t20759 372 17\n",
      "20\t21187\t0.0\t2\t20840 333 14\n",
      "21\t19890\t0.0\t2\t19574 305 11\n",
      "22\t19678\t0.0\t2\t19369 297 12\n",
      "23\t17092\t0.0\t2\t16843 247 2\n",
      "24\t17\t0.0\t2\t12 1 4\n",
      "26\t1\t0.0\t2\t1\n",
      "29\t1\t0.0\t2\t1\n",
      "31\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 16.7%\n",
      "  T: 83.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.8\t1\t0 1\n",
      "129\t1\t0.0\t2\t1\n",
      "131\t1\t0.0\t2\t0 0 1\n",
      "132\t1\t0.0\t2\t1\n",
      "133\t1\t0.0\t2\t1\n",
      "134\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 117296 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.8\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t0 1 1\n",
      "15\t6\t0.0\t2\t6\n",
      "16\t6\t0.0\t2\t6\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t1\t0.0\t2\t0 0 1\n",
      "19\t9\t0.0\t2\t3 4 2\n",
      "20\t99\t0.0\t3\t46 25 19 9\n",
      "21\t21924\t0.0\t3\t20810 814 198 102\n",
      "22\t21229\t0.0\t3\t20095 817 231 86\n",
      "23\t19999\t0.0\t3\t18937 778 184 100\n",
      "24\t19690\t0.0\t3\t18684 763 171 72\n",
      "25\t17339\t0.0\t3\t16358 702 183 96\n",
      "26\t16955\t0.0\t3\t15985 694 190 86\n",
      "27\t23\t0.0\t3\t16 3 1 3\n",
      "56\t1\t0.0\t3\t0 0 0 1\n",
      "59\t1\t0.0\t3\t0 0 1\n",
      "61\t1\t0.0\t3\t1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "63\t1\t0.0\t3\t0 0 1\n",
      "65\t4\t0.0\t3\t0 1 0 3\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S63_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S63_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S63_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S63_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S63_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S63_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S63_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S63_7_L001_R2_001.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.66 s (42 µs/read; 1.42 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             15,750\n",
      "  Read 1 with adapter:                  15,619 (99.2%)\n",
      "  Read 2 with adapter:                  15,464 (98.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              415 (2.6%)\n",
      "Pairs written (passing filters):        15,335 (97.4%)\n",
      "\n",
      "Total basepairs processed:     8,032,500 bp\n",
      "  Read 1:     4,803,750 bp\n",
      "  Read 2:     3,228,750 bp\n",
      "Total written (filtered):      7,146,540 bp (89.0%)\n",
      "  Read 1:     4,360,997 bp\n",
      "  Read 2:     2,785,543 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 38.5%\n",
      "  G: 53.8%\n",
      "  T: 7.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t0.2\t1\t0 4\n",
      "9\t6\t0.1\t1\t0 6\n",
      "20\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "82\t1\t0.0\t3\t1\n",
      "151\t1\t0.0\t3\t1\n",
      "184\t1\t0.0\t3\t1\n",
      "219\t1\t0.0\t3\t1\n",
      "220\t1\t0.0\t3\t1\n",
      "233\t2\t0.0\t3\t2\n",
      "235\t2\t0.0\t3\t1 1\n",
      "236\t3\t0.0\t3\t3\n",
      "237\t2\t0.0\t3\t1 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 15593 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t1\t1\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t6\t0.0\t2\t2 4\n",
      "18\t2446\t0.0\t2\t2402 44\n",
      "19\t2768\t0.0\t2\t2715 52 1\n",
      "20\t2849\t0.0\t2\t2805 44\n",
      "21\t2635\t0.0\t2\t2595 37 3\n",
      "22\t2615\t0.0\t2\t2577 37 1\n",
      "23\t2268\t0.0\t2\t2233 35\n",
      "24\t4\t0.0\t2\t3 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 15464 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t2\t0.0\t2\t2\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t14\t0.0\t3\t7 5 1 1\n",
      "21\t2906\t0.0\t3\t2761 108 28 9\n",
      "22\t2798\t0.0\t3\t2645 109 27 17\n",
      "23\t2515\t0.0\t3\t2362 112 23 18\n",
      "24\t2674\t0.0\t3\t2535 101 25 13\n",
      "25\t2289\t0.0\t3\t2137 111 31 10\n",
      "26\t2259\t0.0\t3\t2108 107 29 15\n",
      "27\t4\t0.0\t3\t4\n",
      "63\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S64_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S64_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S64_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S64_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S64_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S64_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S64_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S64_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.19 s (15 µs/read; 3.88 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            271,003\n",
      "  Read 1 with adapter:                 268,541 (99.1%)\n",
      "  Read 2 with adapter:                 266,287 (98.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,124 (2.6%)\n",
      "Pairs written (passing filters):       263,878 (97.4%)\n",
      "\n",
      "Total basepairs processed:   138,211,530 bp\n",
      "  Read 1:    82,655,915 bp\n",
      "  Read 2:    55,555,615 bp\n",
      "Total written (filtered):    123,011,399 bp (89.0%)\n",
      "  Read 1:    75,078,925 bp\n",
      "  Read 2:    47,932,474 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 175 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.4%\n",
      "  C: 29.7%\n",
      "  G: 65.1%\n",
      "  T: 1.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t28\t4.1\t1\t0 28\n",
      "9\t16\t1.0\t1\t0 16\n",
      "10\t5\t0.3\t1\t0 5\n",
      "20\t5\t0.0\t3\t5\n",
      "21\t13\t0.0\t3\t12 1\n",
      "22\t18\t0.0\t3\t17 0 1\n",
      "23\t13\t0.0\t3\t11 1 1\n",
      "24\t7\t0.0\t3\t6 1\n",
      "25\t15\t0.0\t3\t13 2\n",
      "27\t1\t0.0\t3\t1\n",
      "47\t1\t0.0\t3\t1\n",
      "54\t1\t0.0\t3\t0 1\n",
      "82\t2\t0.0\t3\t2\n",
      "83\t1\t0.0\t3\t1\n",
      "84\t1\t0.0\t3\t1\n",
      "86\t1\t0.0\t3\t1\n",
      "128\t1\t0.0\t3\t1\n",
      "132\t1\t0.0\t3\t1\n",
      "133\t1\t0.0\t3\t1\n",
      "178\t1\t0.0\t3\t1\n",
      "180\t1\t0.0\t3\t1\n",
      "219\t2\t0.0\t3\t2\n",
      "222\t1\t0.0\t3\t1\n",
      "224\t1\t0.0\t3\t0 0 1\n",
      "254\t1\t0.0\t3\t1\n",
      "255\t1\t0.0\t3\t1\n",
      "256\t7\t0.0\t3\t7\n",
      "257\t5\t0.0\t3\t5\n",
      "258\t6\t0.0\t3\t6\n",
      "259\t2\t0.0\t3\t2\n",
      "273\t1\t0.0\t3\t1\n",
      "274\t1\t0.0\t3\t1\n",
      "275\t2\t0.0\t3\t2\n",
      "277\t2\t0.0\t3\t2\n",
      "278\t2\t0.0\t3\t2\n",
      "279\t2\t0.0\t3\t2\n",
      "281\t2\t0.0\t3\t1 1\n",
      "285\t2\t0.0\t3\t2\n",
      "286\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 268366 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t4.1\t1\t3 1\n",
      "9\t1\t1.0\t1\t1\n",
      "10\t4\t0.3\t1\t2 2\n",
      "11\t13\t0.1\t1\t11 2\n",
      "12\t4\t0.0\t1\t3 1\n",
      "13\t5\t0.0\t1\t4 1\n",
      "14\t18\t0.0\t2\t12 3 3\n",
      "15\t16\t0.0\t2\t6 5 5\n",
      "16\t19\t0.0\t2\t7 6 6\n",
      "17\t135\t0.0\t2\t60 50 25\n",
      "18\t42499\t0.0\t2\t41764 699 36\n",
      "19\t48104\t0.0\t2\t47243 803 58\n",
      "20\t48725\t0.0\t2\t47825 865 35\n",
      "21\t45038\t0.0\t2\t44266 715 57\n",
      "22\t45198\t0.0\t2\t44415 746 37\n",
      "23\t38549\t0.0\t2\t37988 552 9\n",
      "24\t31\t0.0\t2\t25 0 6\n",
      "29\t1\t0.0\t2\t1\n",
      "37\t1\t0.0\t2\t1\n",
      "51\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 4 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 25.0%\n",
      "  G: 25.0%\n",
      "  T: 25.0%\n",
      "  none/other: 25.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t4.1\t1\t0 1\n",
      "9\t1\t1.0\t1\t0 1\n",
      "11\t1\t0.1\t1\t0 1\n",
      "205\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 266283 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t1.0\t1\t1 1\n",
      "10\t1\t0.3\t1\t1\n",
      "11\t1\t0.1\t1\t0 1\n",
      "12\t6\t0.0\t1\t5 1\n",
      "14\t3\t0.0\t2\t1 1 1\n",
      "15\t11\t0.0\t2\t8 3\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t5\t0.0\t2\t5\n",
      "18\t10\t0.0\t2\t2 4 4\n",
      "19\t16\t0.0\t2\t4 7 5\n",
      "20\t188\t0.0\t3\t91 44 23 30\n",
      "21\t49341\t0.0\t3\t46737 1913 471 220\n",
      "22\t48099\t0.0\t3\t45639 1843 440 177\n",
      "23\t44929\t0.0\t3\t42569 1758 402 200\n",
      "24\t44950\t0.0\t3\t42581 1771 403 195\n",
      "25\t39727\t0.0\t3\t37455 1665 430 177\n",
      "26\t38931\t0.0\t3\t36642 1706 409 174\n",
      "27\t49\t0.0\t3\t37 7 3 2\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "60\t2\t0.0\t3\t0 1 0 1\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "62\t3\t0.0\t3\t1 0 0 2\n",
      "63\t2\t0.0\t3\t0 0 2\n",
      "64\t2\t0.0\t3\t0 1 1\n",
      "65\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S65_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S65_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S65_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S65_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S65_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S65_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S65_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S65_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.49 s (32 µs/read; 1.88 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             15,466\n",
      "  Read 1 with adapter:                  15,345 (99.2%)\n",
      "  Read 2 with adapter:                  15,209 (98.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              375 (2.4%)\n",
      "Pairs written (passing filters):        15,091 (97.6%)\n",
      "\n",
      "Total basepairs processed:     7,887,660 bp\n",
      "  Read 1:     4,717,130 bp\n",
      "  Read 2:     3,170,530 bp\n",
      "Total written (filtered):      7,033,019 bp (89.2%)\n",
      "  Read 1:     4,291,678 bp\n",
      "  Read 2:     2,741,341 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 27 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 40.7%\n",
      "  G: 59.3%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t0.2\t1\t0 3\n",
      "9\t4\t0.1\t1\t0 4\n",
      "10\t2\t0.0\t1\t0 2\n",
      "20\t2\t0.0\t3\t2\n",
      "22\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t2\t0.0\t3\t2\n",
      "138\t1\t0.0\t3\t1\n",
      "160\t1\t0.0\t3\t1\n",
      "222\t1\t0.0\t3\t1\n",
      "233\t2\t0.0\t3\t2\n",
      "234\t1\t0.0\t3\t1\n",
      "235\t3\t0.0\t3\t3\n",
      "238\t1\t0.0\t3\t1\n",
      "282\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 15318 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t2\n",
      "16\t1\t0.0\t2\t0 1\n",
      "17\t10\t0.0\t2\t5 3 2\n",
      "18\t2401\t0.0\t2\t2359 40 2\n",
      "19\t2702\t0.0\t2\t2660 40 2\n",
      "20\t2796\t0.0\t2\t2747 47 2\n",
      "21\t2564\t0.0\t2\t2515 47 2\n",
      "22\t2645\t0.0\t2\t2611 32 2\n",
      "23\t2191\t0.0\t2\t2159 32\n",
      "24\t4\t0.0\t2\t4\n",
      "47\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 15209 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "19\t3\t0.0\t2\t1 0 2\n",
      "20\t11\t0.0\t3\t5 3 3\n",
      "21\t2805\t0.0\t3\t2649 104 32 20\n",
      "22\t2800\t0.0\t3\t2668 107 17 8\n",
      "23\t2535\t0.0\t3\t2413 88 24 10\n",
      "24\t2568\t0.0\t3\t2428 112 16 12\n",
      "25\t2302\t0.0\t3\t2164 101 27 10\n",
      "26\t2180\t0.0\t3\t2062 84 25 9\n",
      "27\t3\t0.0\t3\t1 1 1\n",
      "63\t1\t0.0\t3\t0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S66_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S66_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S66_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S66_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S66_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S66_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S66_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S66_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.42 s (60 µs/read; 0.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              6,980\n",
      "  Read 1 with adapter:                   6,929 (99.3%)\n",
      "  Read 2 with adapter:                   6,873 (98.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs discarded as untrimmed:              155 (2.2%)\n",
      "Pairs written (passing filters):         6,824 (97.8%)\n",
      "\n",
      "Total basepairs processed:     3,559,800 bp\n",
      "  Read 1:     2,128,900 bp\n",
      "  Read 2:     1,430,900 bp\n",
      "Total written (filtered):      3,181,180 bp (89.4%)\n",
      "  Read 1:     1,941,618 bp\n",
      "  Read 2:     1,239,562 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 4 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 25.0%\n",
      "  G: 75.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.1\t1\t0 1\n",
      "23\t2\t0.0\t3\t2\n",
      "236\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 6925 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t1\t1\n",
      "15\t2\t0.0\t2\t2\n",
      "18\t1074\t0.0\t2\t1062 10 2\n",
      "19\t1217\t0.0\t2\t1196 18 3\n",
      "20\t1294\t0.0\t2\t1261 32 1\n",
      "21\t1206\t0.0\t2\t1185 20 1\n",
      "22\t1151\t0.0\t2\t1121 28 2\n",
      "23\t980\t0.0\t2\t970 9 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 1 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 100.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "205\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 6872 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t7\t0.0\t3\t2 1 2 2\n",
      "21\t1275\t0.0\t3\t1221 42 9 3\n",
      "22\t1226\t0.0\t3\t1185 35 2 4\n",
      "23\t1182\t0.0\t3\t1125 43 10 4\n",
      "24\t1162\t0.0\t3\t1105 41 10 6\n",
      "25\t1022\t0.0\t3\t977 28 11 6\n",
      "26\t995\t0.0\t3\t945 39 7 4\n",
      "27\t2\t0.0\t3\t0 1 0 1\n",
      "59\t1\t0.0\t3\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S68_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S68_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S68_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S68_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S68_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S68_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S68_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S68_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.76 s (50 µs/read; 1.21 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             15,369\n",
      "  Read 1 with adapter:                  15,246 (99.2%)\n",
      "  Read 2 with adapter:                  15,145 (98.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              346 (2.3%)\n",
      "Pairs written (passing filters):        15,023 (97.7%)\n",
      "\n",
      "Total basepairs processed:     7,838,190 bp\n",
      "  Read 1:     4,687,545 bp\n",
      "  Read 2:     3,150,645 bp\n",
      "Total written (filtered):      7,003,039 bp (89.3%)\n",
      "  Read 1:     4,274,098 bp\n",
      "  Read 2:     2,728,941 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 12 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.7%\n",
      "  G: 83.3%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.2\t1\t0 1\n",
      "10\t1\t0.0\t1\t0 1\n",
      "23\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "84\t2\t0.0\t3\t2\n",
      "85\t1\t0.0\t3\t1\n",
      "120\t1\t0.0\t3\t1\n",
      "219\t1\t0.0\t3\t1\n",
      "220\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 15234 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t2\t0.0\t2\t2\n",
      "15\t2\t0.0\t2\t0 0 2\n",
      "17\t15\t0.0\t2\t9 1 5\n",
      "18\t2412\t0.0\t2\t2371 38 3\n",
      "19\t2744\t0.0\t2\t2694 48 2\n",
      "20\t2654\t0.0\t2\t2598 55 1\n",
      "21\t2574\t0.0\t2\t2534 35 5\n",
      "22\t2598\t0.0\t2\t2552 43 3\n",
      "23\t2233\t0.0\t2\t2196 36 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 1 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 100.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "66\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 15144 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.2\t1\t0 1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "15\t1\t0.0\t2\t1\n",
      "18\t1\t0.0\t2\t0 1\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t8\t0.0\t3\t4 3 1\n",
      "21\t2793\t0.0\t3\t2677 87 19 10\n",
      "22\t2749\t0.0\t3\t2617 99 24 9\n",
      "23\t2602\t0.0\t3\t2498 87 14 3\n",
      "24\t2583\t0.0\t3\t2453 97 29 4\n",
      "25\t2218\t0.0\t3\t2105 86 16 11\n",
      "26\t2182\t0.0\t3\t2082 74 17 9\n",
      "27\t2\t0.0\t3\t1 1\n",
      "63\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S69_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S69_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S69_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S69_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S69_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S69_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S69_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S69_19_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.48 s (45 µs/read; 1.34 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             10,791\n",
      "  Read 1 with adapter:                  10,716 (99.3%)\n",
      "  Read 2 with adapter:                  10,597 (98.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              266 (2.5%)\n",
      "Pairs written (passing filters):        10,525 (97.5%)\n",
      "\n",
      "Total basepairs processed:     5,503,410 bp\n",
      "  Read 1:     3,291,255 bp\n",
      "  Read 2:     2,212,155 bp\n",
      "Total written (filtered):      4,906,724 bp (89.2%)\n",
      "  Read 1:     2,995,042 bp\n",
      "  Read 2:     1,911,682 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.7%\n",
      "  G: 83.3%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.0\t1\t1\n",
      "20\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "36\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 10710 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t1\t0.0\t2\t0 1\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t7\t0.0\t2\t5 1 1\n",
      "18\t1753\t0.0\t2\t1714 37 2\n",
      "19\t1852\t0.0\t2\t1815 34 3\n",
      "20\t1883\t0.0\t2\t1849 33 1\n",
      "21\t1921\t0.0\t2\t1881 38 2\n",
      "22\t1759\t0.0\t2\t1730 27 2\n",
      "23\t1529\t0.0\t2\t1513 15 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 10597 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t2\t1\n",
      "20\t8\t0.0\t3\t1 3 3 1\n",
      "21\t1943\t0.0\t3\t1826 95 16 6\n",
      "22\t1853\t0.0\t3\t1743 85 17 8\n",
      "23\t1824\t0.0\t3\t1733 74 10 7\n",
      "24\t1823\t0.0\t3\t1713 79 18 13\n",
      "25\t1612\t0.0\t3\t1524 66 14 8\n",
      "26\t1530\t0.0\t3\t1419 76 21 14\n",
      "27\t3\t0.0\t3\t2 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S71_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S71_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S71_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S71_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S71_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S71_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S71_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S71_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.58 s (55 µs/read; 1.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             10,510\n",
      "  Read 1 with adapter:                  10,402 (99.0%)\n",
      "  Read 2 with adapter:                  10,369 (98.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              249 (2.4%)\n",
      "Pairs written (passing filters):        10,261 (97.6%)\n",
      "\n",
      "Total basepairs processed:     5,360,100 bp\n",
      "  Read 1:     3,205,550 bp\n",
      "  Read 2:     2,154,550 bp\n",
      "Total written (filtered):      4,783,140 bp (89.2%)\n",
      "  Read 1:     2,919,403 bp\n",
      "  Read 2:     1,863,737 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 10 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 100.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t1\t0.0\t3\t1\n",
      "24\t2\t0.0\t3\t0 1 1\n",
      "26\t1\t0.0\t3\t1\n",
      "83\t1\t0.0\t3\t1\n",
      "84\t1\t0.0\t3\t1\n",
      "217\t1\t0.0\t3\t1\n",
      "218\t1\t0.0\t3\t1\n",
      "226\t1\t0.0\t3\t1\n",
      "243\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 10392 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.0\t2\t1\n",
      "16\t2\t0.0\t2\t2\n",
      "17\t6\t0.0\t2\t2 3 1\n",
      "18\t1725\t0.0\t2\t1696 29\n",
      "19\t1882\t0.0\t2\t1856 25 1\n",
      "20\t1910\t0.0\t2\t1876 32 2\n",
      "21\t1768\t0.0\t2\t1739 27 2\n",
      "22\t1638\t0.0\t2\t1597 40 1\n",
      "23\t1460\t0.0\t2\t1440 20\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 10369 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t9\t0.0\t3\t5 2 1 1\n",
      "21\t1933\t0.0\t3\t1839 66 19 9\n",
      "22\t1839\t0.0\t3\t1754 66 13 6\n",
      "23\t1764\t0.0\t3\t1674 68 12 10\n",
      "24\t1728\t0.0\t3\t1624 78 18 8\n",
      "25\t1540\t0.0\t3\t1466 59 12 3\n",
      "26\t1549\t0.0\t3\t1472 57 13 7\n",
      "27\t3\t0.0\t3\t1 2\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S72_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S72_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S72_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S72_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S72_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S72_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S72_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S72_23_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.68 s (24 µs/read; 2.55 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             29,039\n",
      "  Read 1 with adapter:                  28,803 (99.2%)\n",
      "  Read 2 with adapter:                  28,562 (98.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              703 (2.4%)\n",
      "Pairs written (passing filters):        28,336 (97.6%)\n",
      "\n",
      "Total basepairs processed:    14,809,890 bp\n",
      "  Read 1:     8,856,895 bp\n",
      "  Read 2:     5,952,995 bp\n",
      "Total written (filtered):     13,206,456 bp (89.2%)\n",
      "  Read 1:     8,059,567 bp\n",
      "  Read 2:     5,146,889 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 51 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.9%\n",
      "  C: 5.9%\n",
      "  G: 90.2%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.4\t1\t0 1\n",
      "9\t1\t0.1\t1\t0 1\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t3\t0.0\t3\t3\n",
      "22\t3\t0.0\t3\t3\n",
      "23\t3\t0.0\t3\t3\n",
      "24\t4\t0.0\t3\t4\n",
      "25\t5\t0.0\t3\t5\n",
      "40\t1\t0.0\t3\t1\n",
      "62\t2\t0.0\t3\t1 1\n",
      "85\t1\t0.0\t3\t1\n",
      "96\t1\t0.0\t3\t1\n",
      "104\t1\t0.0\t3\t1\n",
      "113\t1\t0.0\t3\t1\n",
      "122\t1\t0.0\t3\t1\n",
      "131\t1\t0.0\t3\t1\n",
      "150\t1\t0.0\t3\t1\n",
      "151\t2\t0.0\t3\t2\n",
      "170\t1\t0.0\t3\t1\n",
      "172\t1\t0.0\t3\t1\n",
      "174\t1\t0.0\t3\t1\n",
      "175\t2\t0.0\t3\t2\n",
      "176\t2\t0.0\t3\t2\n",
      "177\t1\t0.0\t3\t1\n",
      "179\t1\t0.0\t3\t1\n",
      "197\t1\t0.0\t3\t1\n",
      "204\t1\t0.0\t3\t1\n",
      "205\t1\t0.0\t3\t1\n",
      "218\t1\t0.0\t3\t1\n",
      "221\t1\t0.0\t3\t1\n",
      "223\t1\t0.0\t3\t1\n",
      "231\t1\t0.0\t3\t1\n",
      "233\t1\t0.0\t3\t1\n",
      "236\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 28752 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.1\t1\t1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t1\t0.0\t2\t1\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "17\t13\t0.0\t2\t8 3 2\n",
      "18\t4632\t0.0\t2\t4555 75 2\n",
      "19\t5246\t0.0\t2\t5155 86 5\n",
      "20\t5134\t0.0\t2\t5046 87 1\n",
      "21\t4825\t0.0\t2\t4733 91 1\n",
      "22\t4817\t0.0\t2\t4742 69 6\n",
      "23\t4071\t0.0\t2\t3998 70 3\n",
      "24\t6\t0.0\t2\t4 1 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 28562 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "17\t3\t0.0\t2\t1 0 2\n",
      "18\t1\t0.0\t2\t0 0 1\n",
      "19\t1\t0.0\t2\t0 0 1\n",
      "20\t13\t0.0\t3\t6 3 1 3\n",
      "21\t5336\t0.0\t3\t5053 224 37 22\n",
      "22\t5054\t0.0\t3\t4788 202 45 19\n",
      "23\t4835\t0.0\t3\t4578 191 46 20\n",
      "24\t4821\t0.0\t3\t4586 169 46 20\n",
      "25\t4243\t0.0\t3\t4014 164 46 19\n",
      "26\t4247\t0.0\t3\t4005 180 40 22\n",
      "27\t5\t0.0\t3\t3 1 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "66\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S74_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S74_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S74_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S74_25_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S74_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S74_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S74_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S74_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.73 s (25 µs/read; 2.37 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             28,939\n",
      "  Read 1 with adapter:                  28,726 (99.3%)\n",
      "  Read 2 with adapter:                  28,445 (98.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              703 (2.4%)\n",
      "Pairs written (passing filters):        28,236 (97.6%)\n",
      "\n",
      "Total basepairs processed:    14,758,890 bp\n",
      "  Read 1:     8,826,395 bp\n",
      "  Read 2:     5,932,495 bp\n",
      "Total written (filtered):     13,157,876 bp (89.2%)\n",
      "  Read 1:     8,028,750 bp\n",
      "  Read 2:     5,129,126 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 62 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 21.0%\n",
      "  G: 71.0%\n",
      "  T: 8.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t0.4\t1\t0 5\n",
      "9\t6\t0.1\t1\t0 6\n",
      "10\t2\t0.0\t1\t0 2\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t4\t0.0\t3\t3 1\n",
      "22\t2\t0.0\t3\t2\n",
      "24\t2\t0.0\t3\t1 1\n",
      "25\t3\t0.0\t3\t3\n",
      "52\t1\t0.0\t3\t1\n",
      "69\t1\t0.0\t3\t1\n",
      "148\t1\t0.0\t3\t1\n",
      "149\t1\t0.0\t3\t1\n",
      "151\t3\t0.0\t3\t3\n",
      "165\t1\t0.0\t3\t1\n",
      "218\t3\t0.0\t3\t3\n",
      "222\t1\t0.0\t3\t1\n",
      "233\t9\t0.0\t3\t8 1\n",
      "234\t1\t0.0\t3\t1\n",
      "235\t4\t0.0\t3\t3 1\n",
      "236\t1\t0.0\t3\t1\n",
      "237\t5\t0.0\t3\t5\n",
      "238\t1\t0.0\t3\t1\n",
      "252\t1\t0.0\t3\t1\n",
      "260\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 28664 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.1\t1\t1\n",
      "12\t1\t0.0\t1\t0 1\n",
      "14\t2\t0.0\t2\t2\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t9\t0.0\t2\t5 2 2\n",
      "18\t4687\t0.0\t2\t4602 83 2\n",
      "19\t5132\t0.0\t2\t5041 89 2\n",
      "20\t5265\t0.0\t2\t5177 84 4\n",
      "21\t4779\t0.0\t2\t4698 74 7\n",
      "22\t4685\t0.0\t2\t4606 72 7\n",
      "23\t4094\t0.0\t2\t4036 56 2\n",
      "24\t6\t0.0\t2\t5 0 1\n",
      "27\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 4 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.0%\n",
      "  C: 25.0%\n",
      "  G: 0.0%\n",
      "  T: 50.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.4\t1\t0 1\n",
      "9\t2\t0.1\t1\t0 2\n",
      "80\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 28441 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t1\n",
      "15\t2\t0.0\t2\t1 1\n",
      "16\t1\t0.0\t2\t0 0 1\n",
      "18\t1\t0.0\t2\t0 0 1\n",
      "19\t3\t0.0\t2\t3\n",
      "20\t17\t0.0\t3\t5 5 4 3\n",
      "21\t5314\t0.0\t3\t5040 214 44 16\n",
      "22\t5126\t0.0\t3\t4840 210 53 23\n",
      "23\t4805\t0.0\t3\t4562 178 41 24\n",
      "24\t4817\t0.0\t3\t4520 213 63 21\n",
      "25\t4212\t0.0\t3\t3954 197 39 22\n",
      "26\t4132\t0.0\t3\t3879 183 55 15\n",
      "27\t6\t0.0\t3\t6\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "65\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S82_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S82_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S82_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S82_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S82_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S82_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S82_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S82_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.33 s (19 µs/read; 3.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             71,899\n",
      "  Read 1 with adapter:                  71,267 (99.1%)\n",
      "  Read 2 with adapter:                  70,823 (98.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,695 (2.4%)\n",
      "Pairs written (passing filters):        70,204 (97.6%)\n",
      "\n",
      "Total basepairs processed:    36,668,490 bp\n",
      "  Read 1:    21,929,195 bp\n",
      "  Read 2:    14,739,295 bp\n",
      "Total written (filtered):     32,730,931 bp (89.3%)\n",
      "  Read 1:    19,977,855 bp\n",
      "  Read 2:    12,753,076 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 15 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 33.3%\n",
      "  G: 66.7%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t1.1\t1\t0 2\n",
      "9\t3\t0.3\t1\t0 3\n",
      "20\t3\t0.0\t3\t3\n",
      "21\t1\t0.0\t3\t1\n",
      "23\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "26\t1\t0.0\t3\t1\n",
      "217\t1\t0.0\t3\t1\n",
      "218\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 71252 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.3\t1\t0 1\n",
      "10\t1\t0.1\t1\t1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t3\t0.0\t1\t2 1\n",
      "13\t1\t0.0\t1\t1\n",
      "15\t8\t0.0\t2\t5 2 1\n",
      "16\t5\t0.0\t2\t2 0 3\n",
      "17\t26\t0.0\t2\t12 12 2\n",
      "18\t11340\t0.0\t2\t11124 207 9\n",
      "19\t12930\t0.0\t2\t12699 221 10\n",
      "20\t12930\t0.0\t2\t12687 234 9\n",
      "21\t12014\t0.0\t2\t11806 197 11\n",
      "22\t12106\t0.0\t2\t11892 207 7\n",
      "23\t9875\t0.0\t2\t9696 176 3\n",
      "24\t10\t0.0\t2\t9 0 1\n",
      "42\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 1 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 100.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "161\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 70822 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t1\t0.0\t1\t1\n",
      "15\t4\t0.0\t2\t2 2\n",
      "17\t4\t0.0\t2\t3 0 1\n",
      "18\t1\t0.0\t2\t1\n",
      "19\t3\t0.0\t2\t0 1 2\n",
      "20\t56\t0.0\t3\t28 20 3 5\n",
      "21\t13284\t0.0\t3\t12705 441 94 44\n",
      "22\t12816\t0.0\t3\t12236 437 88 55\n",
      "23\t11948\t0.0\t3\t11406 414 82 46\n",
      "24\t11964\t0.0\t3\t11391 442 85 46\n",
      "25\t10481\t0.0\t3\t10016 359 70 36\n",
      "26\t10245\t0.0\t3\t9688 399 110 48\n",
      "27\t11\t0.0\t3\t8 2 1\n",
      "28\t1\t0.0\t3\t1\n",
      "58\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S83_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S83_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S83_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S83_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S83_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S83_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S83_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S83_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.71 s (20 µs/read; 3.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             35,572\n",
      "  Read 1 with adapter:                  35,245 (99.1%)\n",
      "  Read 2 with adapter:                  35,051 (98.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              843 (2.4%)\n",
      "Pairs written (passing filters):        34,729 (97.6%)\n",
      "\n",
      "Total basepairs processed:    18,141,720 bp\n",
      "  Read 1:    10,849,460 bp\n",
      "  Read 2:     7,292,260 bp\n",
      "Total written (filtered):     16,190,591 bp (89.2%)\n",
      "  Read 1:     9,881,513 bp\n",
      "  Read 2:     6,309,078 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 23 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 21.7%\n",
      "  G: 78.3%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t0.5\t1\t0 2\n",
      "9\t3\t0.1\t1\t0 3\n",
      "20\t3\t0.0\t3\t1 1 1\n",
      "22\t2\t0.0\t3\t2\n",
      "23\t2\t0.0\t3\t2\n",
      "25\t1\t0.0\t3\t1\n",
      "80\t1\t0.0\t3\t1\n",
      "81\t2\t0.0\t3\t2\n",
      "92\t1\t0.0\t3\t1\n",
      "170\t1\t0.0\t3\t1\n",
      "208\t1\t0.0\t3\t1\n",
      "218\t1\t0.0\t3\t1\n",
      "219\t1\t0.0\t3\t1\n",
      "274\t1\t0.0\t3\t1\n",
      "275\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 35222 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.1\t1\t1\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t1\t0.0\t1\t0 1\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t3\t0.0\t2\t2 0 1\n",
      "16\t2\t0.0\t2\t2\n",
      "17\t22\t0.0\t2\t14 5 3\n",
      "18\t5610\t0.0\t2\t5524 80 6\n",
      "19\t6293\t0.0\t2\t6197 91 5\n",
      "20\t6441\t0.0\t2\t6331 108 2\n",
      "21\t6011\t0.0\t2\t5914 91 6\n",
      "22\t5864\t0.0\t2\t5775 88 1\n",
      "23\t4968\t0.0\t2\t4890 77 1\n",
      "24\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 35051 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "17\t1\t0.0\t2\t1\n",
      "20\t32\t0.0\t3\t12 12 1 7\n",
      "21\t6701\t0.0\t3\t6450 200 29 22\n",
      "22\t6254\t0.0\t3\t5982 214 42 16\n",
      "23\t5981\t0.0\t3\t5716 188 44 33\n",
      "24\t5804\t0.0\t3\t5550 192 43 19\n",
      "25\t5194\t0.0\t3\t4962 177 38 17\n",
      "26\t5074\t0.0\t3\t4841 185 31 17\n",
      "27\t9\t0.0\t3\t5 3 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S84_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S84_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S84_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S84_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S84_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S84_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S84_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S84_9_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.90 s (20 µs/read; 2.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             44,482\n",
      "  Read 1 with adapter:                  44,118 (99.2%)\n",
      "  Read 2 with adapter:                  43,780 (98.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,059 (2.4%)\n",
      "Pairs written (passing filters):        43,423 (97.6%)\n",
      "\n",
      "Total basepairs processed:    22,685,820 bp\n",
      "  Read 1:    13,567,010 bp\n",
      "  Read 2:     9,118,810 bp\n",
      "Total written (filtered):     20,243,998 bp (89.2%)\n",
      "  Read 1:    12,356,229 bp\n",
      "  Read 2:     7,887,769 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 14 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 35.7%\n",
      "  G: 64.3%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.7\t1\t0 1\n",
      "9\t3\t0.2\t1\t0 3\n",
      "10\t1\t0.0\t1\t0 1\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t3\t0.0\t3\t3\n",
      "47\t1\t0.0\t3\t1\n",
      "81\t1\t0.0\t3\t1\n",
      "100\t1\t0.0\t3\t1\n",
      "164\t1\t0.0\t3\t1\n",
      "171\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 44104 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.0\t1\t1\n",
      "11\t3\t0.0\t1\t3\n",
      "13\t3\t0.0\t1\t2 1\n",
      "15\t4\t0.0\t2\t3 1\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t23\t0.0\t2\t11 7 5\n",
      "18\t6855\t0.0\t2\t6738 102 15\n",
      "19\t7908\t0.0\t2\t7782 116 10\n",
      "20\t8154\t0.0\t2\t8007 142 5\n",
      "21\t7546\t0.0\t2\t7415 119 12\n",
      "22\t7385\t0.0\t2\t7269 112 4\n",
      "23\t6212\t0.0\t2\t6114 95 3\n",
      "24\t6\t0.0\t2\t3 2 1\n",
      "26\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 0 times\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 43780 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t0 0 1\n",
      "15\t2\t0.0\t2\t2\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t1\t0.0\t2\t0 0 1\n",
      "19\t3\t0.0\t2\t2 1\n",
      "20\t32\t0.0\t3\t12 10 4 6\n",
      "21\t8120\t0.0\t3\t7786 249 65 20\n",
      "22\t7857\t0.0\t3\t7499 256 62 40\n",
      "23\t7377\t0.0\t3\t7032 247 65 33\n",
      "24\t7452\t0.0\t3\t7086 272 61 33\n",
      "25\t6555\t0.0\t3\t6228 251 46 30\n",
      "26\t6373\t0.0\t3\t6027 262 55 29\n",
      "27\t3\t0.0\t3\t3\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S85_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S85_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S85_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S85_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S85_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S85_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S85_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S85_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.17 s (20 µs/read; 3.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             58,322\n",
      "  Read 1 with adapter:                  57,822 (99.1%)\n",
      "  Read 2 with adapter:                  57,434 (98.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,379 (2.4%)\n",
      "Pairs written (passing filters):        56,943 (97.6%)\n",
      "\n",
      "Total basepairs processed:    29,744,220 bp\n",
      "  Read 1:    17,788,210 bp\n",
      "  Read 2:    11,956,010 bp\n",
      "Total written (filtered):     26,546,948 bp (89.3%)\n",
      "  Read 1:    16,202,424 bp\n",
      "  Read 2:    10,344,524 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 31 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 19.4%\n",
      "  G: 77.4%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t0.9\t1\t0 3\n",
      "9\t1\t0.2\t1\t0 1\n",
      "10\t1\t0.1\t1\t0 1\n",
      "21\t3\t0.0\t3\t3\n",
      "22\t6\t0.0\t3\t6\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t2\t0.0\t3\t2\n",
      "50\t1\t0.0\t3\t1\n",
      "72\t1\t0.0\t3\t1\n",
      "73\t1\t0.0\t3\t1\n",
      "80\t2\t0.0\t3\t2\n",
      "85\t1\t0.0\t3\t1\n",
      "148\t1\t0.0\t3\t1\n",
      "215\t1\t0.0\t3\t1\n",
      "219\t1\t0.0\t3\t1\n",
      "238\t1\t0.0\t3\t1\n",
      "285\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 57791 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t1\t1 1\n",
      "11\t1\t0.0\t1\t1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t3\t0.0\t2\t2 1\n",
      "15\t7\t0.0\t2\t4 1 2\n",
      "16\t8\t0.0\t2\t2 3 3\n",
      "17\t23\t0.0\t2\t11 12\n",
      "18\t9042\t0.0\t2\t8878 155 9\n",
      "19\t10303\t0.0\t2\t10114 181 8\n",
      "20\t10624\t0.0\t2\t10429 185 10\n",
      "21\t9757\t0.0\t2\t9572 177 8\n",
      "22\t9876\t0.0\t2\t9708 162 6\n",
      "23\t8135\t0.0\t2\t7997 134 4\n",
      "24\t7\t0.0\t2\t7\n",
      "25\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 2 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 50.0%\n",
      "  C: 50.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.9\t1\t0 1\n",
      "204\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 57432 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t4\t0.0\t2\t3 1\n",
      "16\t1\t0.0\t2\t0 0 1\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t5\t0.0\t2\t5\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t40\t0.0\t3\t13 12 7 8\n",
      "21\t10822\t0.0\t3\t10379 341 67 35\n",
      "22\t10443\t0.0\t3\t9973 354 84 32\n",
      "23\t9802\t0.0\t3\t9409 296 72 25\n",
      "24\t9531\t0.0\t3\t9098 310 91 32\n",
      "25\t8530\t0.0\t3\t8135 282 85 28\n",
      "26\t8233\t0.0\t3\t7817 333 53 30\n",
      "27\t15\t0.0\t3\t11 3 0 1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "60\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S86_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S86_29_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S86_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S86_29_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S86_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ehp71vq9/S86_29_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S86_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-j6h5lrr4/0000f8ed-d420-4fca-bc8c-7f2da1cd7718/data/S86_29_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.91 s (20 µs/read; 2.98 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             45,210\n",
      "  Read 1 with adapter:                  44,840 (99.2%)\n",
      "  Read 2 with adapter:                  44,467 (98.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,109 (2.5%)\n",
      "Pairs written (passing filters):        44,101 (97.5%)\n",
      "\n",
      "Total basepairs processed:    23,057,100 bp\n",
      "  Read 1:    13,789,050 bp\n",
      "  Read 2:     9,268,050 bp\n",
      "Total written (filtered):     20,558,287 bp (89.2%)\n",
      "  Read 1:    12,547,479 bp\n",
      "  Read 2:     8,010,808 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.0%\n",
      "  C: 12.0%\n",
      "  G: 84.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t0.7\t1\t0 2\n",
      "10\t1\t0.0\t1\t0 1\n",
      "21\t1\t0.0\t3\t0 1\n",
      "24\t3\t0.0\t3\t2 1\n",
      "25\t1\t0.0\t3\t1\n",
      "70\t1\t0.0\t3\t1\n",
      "71\t1\t0.0\t3\t1\n",
      "75\t1\t0.0\t3\t1\n",
      "81\t1\t0.0\t3\t1\n",
      "83\t1\t0.0\t3\t1\n",
      "85\t1\t0.0\t3\t1\n",
      "147\t1\t0.0\t3\t1\n",
      "150\t1\t0.0\t3\t1\n",
      "151\t1\t0.0\t3\t1\n",
      "214\t1\t0.0\t3\t1\n",
      "218\t2\t0.0\t3\t2\n",
      "220\t3\t0.0\t3\t3\n",
      "243\t1\t0.0\t3\t1\n",
      "286\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 44815 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t0 2\n",
      "15\t2\t0.0\t2\t1 1\n",
      "17\t33\t0.0\t2\t18 12 3\n",
      "18\t7098\t0.0\t2\t6987 105 6\n",
      "19\t8146\t0.0\t2\t7994 147 5\n",
      "20\t8158\t0.0\t2\t8012 139 7\n",
      "21\t7476\t0.0\t2\t7354 114 8\n",
      "22\t7542\t0.0\t2\t7408 130 4\n",
      "23\t6354\t0.0\t2\t6257 95 2\n",
      "24\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 1 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 100.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 44466 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.0\t2\t0 1\n",
      "15\t4\t0.0\t2\t4\n",
      "17\t1\t0.0\t2\t1\n",
      "18\t1\t0.0\t2\t1\n",
      "19\t2\t0.0\t2\t0 0 2\n",
      "20\t30\t0.0\t3\t11 8 2 9\n",
      "21\t8283\t0.0\t3\t7938 255 64 26\n",
      "22\t7939\t0.0\t3\t7585 265 62 27\n",
      "23\t7481\t0.0\t3\t7126 270 61 24\n",
      "24\t7567\t0.0\t3\t7195 281 61 30\n",
      "25\t6678\t0.0\t3\t6326 267 57 28\n",
      "26\t6472\t0.0\t3\t6162 236 52 22\n",
      "27\t6\t0.0\t3\t5 1\n",
      "28\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = trim_paired(\n",
    "        demultiplexed_sequences=artifact,\n",
    "        front_f=forward_primer,\n",
    "        front_r=reverse_primer,\n",
    "        adapter_f=reverse_reverse_complement,\n",
    "        adapter_r=forward_reverse_complement,\n",
    "        cores=threads,\n",
    "        overlap=trim['overlap'],\n",
    "        indels=False,\n",
    "        match_read_wildcards=True,\n",
    "        match_adapter_wildcards=True,\n",
    "        error_rate=0.15,\n",
    "        discard_untrimmed=True,\n",
    "    ).trimmed_sequences\n",
    "\n",
    "    demux.visualizers.summarize(res).visualization\n",
    "    res.save(demux_file_trim)\n",
    "    Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 89.588728,
   "end_time": "2022-11-29T18:42:22.266753",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-andressa-lbtm/experiments/andressa-01/nb-executed-steps/step-prepare-data-andressa-01.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-andressa-lbtm",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "andressa",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-andressa-lbtm/data/manifest.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-andressa-lbtm/data/metadata.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2022-11-29T18:40:52.678025",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}